platform-sdk/platform-apps/tests/ISSTestingTool/src/main/java/com/swirlds/demo/iss/ISSTestingToolState.java

```diff
@@ -292,14 +304,43 @@ private void captureTimestamp(final ConsensusEvent event) {
      *
      * @param transaction the transaction to apply
      */
-    private void handleTransaction(final ConsensusTransaction transaction) {
-        if (transaction.isSystem()) {
-            return;
+    private ConsensusTransaction handleTransaction(final ConsensusTransaction transaction) {
+        if (isSystemTransaction(transaction)) {
+            return transaction;
         }
+
         final int delta =
                 ByteUtils.byteArrayToInt(transaction.getApplicationTransaction().toByteArray(), 0);
         runningSum += delta;
         setChild(RUNNING_SUM_INDEX, new StringLeaf(Long.toString(runningSum)));
+
+        return null;
+    }
+
+    private boolean isSystemTransaction(final ConsensusTransaction transaction) {
+        if (transaction.isSystem()) {
```

Comment:
This method will not be available once we switch to the new transaction structure. If you drill down into it, it depends on the `oneof` protobuf structure in `EventTransaction` that holds either an application transaction or a state signature transaction. This entire effort is to get rid of `EventTransaction` and the need for the `oneof`.

The `encodeSystemTransaction()` method added in the main class must introduce a new way of distinguishing between a system transaction and it's own application transaction in the serialized bytes returned. For example, a simple option is to prepend the bytes of the system transaction with a `0` to indicate a system transaction, and prepend application transactions with a `1`. That way, when you get to this method and you have to determine if the `Bytes` inside the `ConsensusTransaction` represent an application transaction or a system transaction.

Reply:
I know this method will not be available once we switch to the new type. However, we discussed for some time that we should strive to support both the old and the new format. I know this method depends on the `oneof` protobuf structure. What's the last direction we should be going with? We completely remove the EventTransaction support, so we should `only` support the new format? This would mean all changes should go in a common single milestone.

About the prepend of 0 and 1 bytes we have discussed with @lpetrovic05 that having 0 or 1 as a distinguish factor will not be enough and we should use real StateSignatureTransaction. Having random bytes for an application transaction can produce transactions starting with both 0 or 1 bytes. The discussion can be seen [here](https://github.com/hashgraph/hedera-services/pull/16844#discussion_r1871408813).

Reply:
this method will not be available in the future, but its easier to leave it for now until the migration is complete